File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int low = 0;
        int high = nums.size() - 1;

        while (low < high) {
            
            if (nums[low] + nums[high] == target) {
                return {low + 1, high + 1};
            } 
            else if (nums[low] + nums[high] < target) {
                low++;
            } 
            else {
                high--;
            }
        }
        return {-1, -1};
    }
};

File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        #first using binary search
        left = 0
        right = len(numbers) - 1

        while left < right:
            mid = left + (right - left) // 2
            if numbers[left] + numbers[right] == target:
                return [left + 1, right + 1]
            elif numbers[left] + numbers[right] > target:
                right -= 1
            else:
                left += 1
        return [-1, -1]


        #using dictionary
        # ans = {}
        # for i in range(len(numbers)):
        #     el = target - numbers[i]
        #     if el in ans:
        #         return [ans[el] + 1, i + 1]
        #     ans[numbers[i]] = i
        # return [-1, -1]
